src/Test/Utilities/Portable/Generated.cs

```diff
@@ -0,0 +1,167 @@
+        private static byte[] _SystemThreadingTasks;
+        public static byte[] SystemThreadingTasks => ResourceLoader.GetOrCreateResource(ref _SystemThreadingTasks, "net461.System.Threading.Tasks.dll");
+        private static byte[] _MicrosoftCSharp;
+        public static byte[] MicrosoftCSharp => ResourceLoader.GetOrCreateResource(ref _MicrosoftCSharp, "net461.Microsoft.CSharp.dll");
+        private static byte[] _MicrosoftVisualBasic;
+        public static byte[] MicrosoftVisualBasic => ResourceLoader.GetOrCreateResource(ref _MicrosoftVisualBasic, "net461.Microsoft.VisualBasic.dll");
+    }
+    public static class Net461
+    {
+        public static PortableExecutableReference mscorlib { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.mscorlib).GetReference(display: "mscorlib.dll (net461)");
+        public static PortableExecutableReference System { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.System).GetReference(display: "System.dll (net461)");
+        public static PortableExecutableReference SystemCore { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.SystemCore).GetReference(display: "System.Core.dll (net461)");
+        public static PortableExecutableReference SystemRuntime { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.SystemRuntime).GetReference(display: "System.Runtime.dll (net461)");
+        public static PortableExecutableReference SystemThreadingTasks { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.SystemThreadingTasks).GetReference(display: "System.Threading.Tasks.dll (net461)");
+        public static PortableExecutableReference MicrosoftCSharp { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.MicrosoftCSharp).GetReference(display: "Microsoft.CSharp.dll (net461)");
+        public static PortableExecutableReference MicrosoftVisualBasic { get; } = AssemblyMetadata.CreateFromImage(ResourcesNet461.MicrosoftVisualBasic).GetReference(display: "Microsoft.VisualBasic.dll (net461)");
+    }
+    public static class ResourcesValueTask
+    {
+        private static byte[] _SystemThreadingTasksExtensions;
+        public static byte[] SystemThreadingTasksExtensions => ResourceLoader.GetOrCreateResource(ref _SystemThreadingTasksExtensions, "ValueTask.System.Threading.Tasks.Extensions.dll");
+    }
+    public static class ValueTask
+    {
+        public static PortableExecutableReference SystemThreadingTasksExtensions { get; } = AssemblyMetadata.CreateFromImage(ResourcesValueTask.SystemThreadingTasksExtensions).GetReference(display: "System.Threading.Tasks.Extensions.dll (ValueTask)");
+    }
+    public static class ResourcesBuildExtensions
+    {
+        private static byte[] _NetStandard461;
+        public static byte[] NetStandard461 => ResourceLoader.GetOrCreateResource(ref _NetStandard461, "BuildExtensions.netstandard.dll");
+    }
+    public static class BuildExtensions
```

Comment:
>BuildExtensions [](start = 24, length = 15)

This name does not indicate that it's a PE reference created for metadata in a resource.

Reply:
Have a better suggestion here? I'd prefer one that we could apply as a pattern to these one off reference cases. There are two right now but looking forward there will likely be 1-2 more. Hence if we had a nice prefix or suffix it could be applied to all of them.

Reply:
As noted above I'd encapsulate all of these types in something like `TestMetadata` static class. Then the names are ok as nested types.